provider/provider.go

```diff
@@ -1,18 +1,18 @@
-// Package provider implements structures and methods to provide blocks,
-// keep track of which blocks are provided, and to allow those blocks to
-// be reprovided.
 package provider
 
 import (
 	"context"
 	"github.com/ipfs/go-cid"
 	logging "github.com/ipfs/go-log"
-	"github.com/libp2p/go-libp2p-routing"
 )
 
-var log = logging.Logger("provider")
+var (
+	// StrategicProvidingEnabled toggles between the original providing mechanism
+	// and the new strategic providing system
+	StrategicProvidingEnabled = false
```

Comment:
I'd avoid using global vars to configure subsystems, it tends to make some things painful (parallel go tests to name one example)

Reply:
I just want to clarify here that this was an artifact from an early version and I just removed it. The experimental subsystem configuration is done by constructing different types of objects.